qingstor/sdk/request.py

```diff
@@ -107,6 +107,11 @@ def get_canonicalized_resource(self):
                     else:
                         keys.append(i)
             keys = sorted(keys)
+        fileds = parsed_uri.hostname.split(".")
+        if len(fileds) < 4:
```

Comment:
Add a comment about what `parsed_uri.hostname` looks like can make code much readable.

Reply:
Do we have a more reliable detect method here?

For example, what happened if our hostname is `qingstor.object.storage.xuanwo.io`?